automaxlair/matchup_scoring.py

```diff
@@ -46,6 +47,104 @@ def type_damage_multiplier(move_type: str, defender_types: List[str]) -> float:
     return factor
 
 
+def weather_damage_multiplier(
+    attacker: Pokemon, move: Move, defender: Pokemon, field: Field
+) -> float:
+    modifier = 1
+    if field.is_weather_sunlight():
+        if move.name_id == 'solar-beam':
+            modifier *= 2
+        elif move.name_id == 'thunder' or move.name_id == 'hurricane':
+            modifier *= (0.5 / 0.7)
+        elif move.name_id == 'weather-ball':
+            move.type_id = 'fire'
+            modifier *= 2
+
+        if move.type_id == 'fire':
+            modifier *= 1.5
+        elif move.type_id == 'water':
+            modifier *= 0.5
+    elif field.is_weather_rain():
+        if move.name_id == 'solar-beam':
+            modifier *= 0.5
+        elif move.name_id == 'thunder' or move.name_id == 'hurricane':
+            modifier *= (1.0 / 0.7)
+        elif move.name_id == 'weather-ball':
+            move.type_id = 'water'
+            modifier *= 2
+
+        if move.type_id == 'water':
+            modifier *= 1.5
+        elif move.type_id == 'fire':
+            modifier *= 0.5
+
+    elif field.is_weather_hail():
+        if move.name_id == 'solar-beam':
+            modifier *= 0.5
+        elif move.name_id == 'blizzard':
+            modifier *= (1.0 / 0.7)
+        elif move.name_id == 'weather-ball':
+            move.type_id = 'ice'
+            modifier *= 2
+    elif field.is_weather_sandstorm():
+        if move.name_id == 'solar-beam':
+            modifier *= 0.5
+        elif move.name_id == 'weather-ball':
+            move.type_id = 'rock'
+            modifier *= 2
+    else:
+        if move.name_id == 'weather-ball':
+            move.type_id = 'normal'
```

Comment:
What happens when the battle ends with weather? Does Weather Ball's typing stay different then? I wonder if that could affect viability calculations.

Reply:
indeed, it's not reseted. Let me do that. It would have done that not only for weather ball (because the current_run is resetted when we being a new DA and not a new fight)

Reply:
well  actually i lied

>     def reset_stage(self) -> None:
>         """Reset after a battle."""
>         self.field = Field()
so it will reset